src/axom/mir/MeshTester.cpp

```diff
@@ -82,6 +83,118 @@ static axom::float64 calculatePercentOverlapMonteCarlo(int gridSize,
   }
 }
 
+//--------------------------------------------------------------------------------
+template <typename TopologyView, typename CoordsetView>
+void generateSphericalVolumeFractions(
+  TopologyView topologyView,
+  CoordsetView coordsetView,
+  int gridSize,
+  int numSpheres,
+  std::vector<std::vector<axom::float64>>& materialVolumeFractionsData)
+{
+  using PointType = typename CoordsetView::PointType;
+
+  // Generate the element volume fractions with concentric spheres
+  int numMaterials = numSpheres + 1;
+  int defaultMaterialID =
+    numMaterials - 1;  // default material is always the last index
+
+  // Initialize the radii of the circles
+  std::vector<axom::float64> sphereRadii;
+  axom::float64 maxRadius =
+    gridSize / 2.0;  // Note: The choice of divisor is arbitrary
+  axom::float64 minRadius =
+    gridSize / 4.0;  // Note: The choice of divisor is arbitrary
+
+  axom::float64 radiusDelta;
+  if(numSpheres <= 1)
+    radiusDelta = (maxRadius - minRadius);
+  else
+    radiusDelta =
+      (maxRadius - minRadius) / static_cast<axom::float64>(numSpheres - 1);
+
+  for(int i = 0; i < numSpheres; ++i)
+  {
+    auto rad = minRadius + (i * radiusDelta);
+    sphereRadii.push_back(rad * rad);
+  }
+
+  // Initialize all material volume fractions to 0
+  materialVolumeFractionsData.resize(numMaterials);
+  for(int i = 0; i < numMaterials; ++i)
+  {
+    materialVolumeFractionsData[i].resize(topologyView.numberOfZones(), 0.);
+  }
+
+  // all spheres are centered around the same point
+  const float c = static_cast<float>(gridSize / 2.0);
+  const auto sphereCenter = PointType({c, c, c});
```

Comment:
would it make sense to use `primal::Sphere()` here?

(I thought we had `Sphere::inside()`, but it looks like we have `Sphere::computeSignedDistance(pt) < 0`

Reply:
I substituted a newer version of the method that I rewrote on my 3D branch. It still does not use Sphere but I think that's ok.